src/components/Admins/index.js

```diff
@@ -24,82 +23,87 @@ const Admins = () => {
-    setIsOpen(true);
-  }
+  const handleDeleteAdmin = (admin) => {
+    dispatch(deleteAdmin(admin));
+  };
 
-  function setBothDatas(data) {
-    setData(data);
-    setUntouchedData(data);
+  if (loading) {
+    return (
+      <div className={styles.loading}>
+        <LoadingScreen />
+      </div>
+    );
   }
 
-  function search(value) {
+  const search = (value) => {
     setSearchQuery(value);
-    const result = untouchedData.filter((employee) => employee._id.includes(value));
-    setData(result);
-  }
-
-  async function deleteAdmin(id) {
-    const response = await fetch(`${URL}/${id}`, { method: 'DELETE' });
-    const responseJson = await response.json();
-    if (responseJson.error) {
-      alert('error');
-    } else {
-      setBothDatas(data.filter((item) => item._id !== id));
-      setIsOpen(false);
-    }
-  }
+    setFilteredList(admins.filter((employee) => employee._id.includes(searchQuery)));
+  };
 
   return (
     <section className={styles.container}>
-      {loading ? (
+      <Modal
+        modalTitle={'Admins'}
+        isOpen={isOpen}
+        handleClose={() => {
+          setIsOpen(!isOpen);
+        }}
+      >
+        <p>{error ? error : 'Are you sure to delete an admin?'}</p>
         <div>
-          <LoadingScreen />
-        </div>
-      ) : (
-        <>
-          <Modal
-            modalTitle="Admin Delete"
-            isOpen={isOpen}
-            handleClose={() => {
-              setIsOpen(!isOpen);
-            }}
-          >
+          {error ? (
+            <div>
+              <Button text="Close" handler={() => setIsOpen(false)} />
+            </div>
+          ) : (
             <div>
-              <p>Are you sure you want to delete admin?</p>
+              <Button
+                text="Yes"
+                type="delete"
+                handler={() => {
+                  handleDeleteAdmin(idDel);
+                  setIsOpen(!isOpen);
+                }}
+              />
+              <Button text="No" handler={() => setIsOpen(false)} />
             </div>
-            <Button text="Yes" type="delete" handler={() => deleteAdmin(deleteId)} />
-            <Button text="No" handler={() => setIsOpen(false)} />
-          </Modal>
-          <h2>Admins</h2>
-          <div className={styles.buttonContainer}>
-            <Button text="Add Admin" link={'/admins/form'} />
-            <Search placeholder="Search admin" searchQuery={searchQuery} setSearchQuery={search} />
-          </div>
-          <Table data={data} column={column} deleteItem={handleDeleteAdmin} entity="admins" />
-        </>
-      )}
+          )}
+        </div>
+      </Modal>
+      <h2>Admins</h2>
+      <div className={styles.buttonContainer}>
+        <Button text={'Add Admin'} link={'/admins/form'} />
+        <Search searchQuery={searchQuery} setSearchQuery={search} placeholder={'Search admin'} />
+      </div>
+      <Table
+        data={searchQuery.length == 0 ? admins : filteredList}
```

Comment:
A better way of doing this condition is:
```suggestion
        data={searchQuery.length ? filteredList : admins}
```

Because on binary, 0 means `false` and any other number > 0 is `true`